### PR TITLE
142558933 post johari aw dp

### PR DESCRIPTION
**Pull request by Dpalazzari**
In this PR:

- adds ability to POST to /api/v1/users/:id/descriptions
- Adds request test for Post
- Adds model tests
- All tests pass

[#142558933]

FYI @case-eee

### CODE REVIEW

File: spec/requests/api/v1/users/user_descriptions_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    create(:adjective, name: 'cool')
+    user1, user2 = create_list(:user, 2)
+    assignment = create(:assignment, assignee: user1, assigned: user2)
+    
+    expect(user2.descriptions.count).to eq(0)
+
+    adjectives = ['shy', 'religious', 'cool']
+    headers = { 'CONTENT_TYPE' => 'application/json' }
+    json = {:johari => adjectives, :describer_id => user1.id }
+
```
Comment (lucyconklin) [inline spec/requests/api/v1/users/user_descriptions_spec.rb]: 👍

**Comment by lucyconklin** [comment]
LGTM !

File: app/controllers/api/v1/users/descriptions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Api::V1::Users::DescriptionsController < Api::V1::BaseController
+  
+  def create
+    describee = User.find(params[:id])
+    descriptions = JSON.parse(request.body.read)
+    describer    = User.find(descriptions['describer_id'])
+    describee.create_descriptions(descriptions['johari'], describer)
+    if describee.all_descriptions_completed?(describer.id, descriptions['johari'])
+      Assignment.complete(describee.id, describer.id)
```
Comment (kheppenstall) [inline app/controllers/api/v1/users/descriptions_controller.rb]: @Dpalazzari @wlffann What if an assignment has already been completed or has not been assigned?

Are we checking to make sure there is an existing assignment for the evaluator and evaluatee and that the assignment has not already been completed before we all the post request to go through?

This seems like the right place to make sure not anyone can submit descriptions for anyone else (or make multiple posts) using something like Postman (unless when we're already doing that and I just can't read code good). 

I'm trying to follow along but thought I'd just ask.

Comment (Dpalazzari) [inline app/controllers/api/v1/users/descriptions_controller.rb]: @kheppenstall I don't think we are checking that. Postman won't be a problem once we secure the API. Maybe make a card for this?

Comment (kheppenstall) [inline app/controllers/api/v1/users/descriptions_controller.rb]: @Dpalazzari Yea that makes sense with the postman. My other thought is someone just manually entering the url as `johari/:id` and resubmitting (like if they submitted and then went back to the page and then submitted again). So I guess higher priority would be that we check there is an open assignment before letting any descriptions be added.

Comment (Dpalazzari) [inline app/controllers/api/v1/users/descriptions_controller.rb]: Yeah that shouldn't be too hard. just have to find the assignments where completed? is false

Comment (kheppenstall) [inline app/controllers/api/v1/users/descriptions_controller.rb]: Yea that makes sense. Added the story.

https://www.pivotaltracker.com/n/projects/1996573

**Pull request merged**